src/shogun/machine/Machine.h

```diff
@@ -25,6 +25,20 @@
 namespace shogun
 {
 
+#define TRAIN_DENSE_DISPATHER(train_templated_name) \
+virtual bool train_with_dense(CFeatures* data) \
+{  \
+  switch (data->get_feature_type()) \
+  { \
+    case F_DREAL: \
+      return train_templated_name<float64_t>(data->as<CDenseFeatures<float64_t>>()); \
```

Comment:
I guess we could get away with static casting here as well, but I am always more comfortable using a dynamic cast as done in  `as`